### PR TITLE
Send completion info from DebugLinkDecisions

### PR DESCRIPTION
**Pull request by matejcik**
This allows us to get rid of sleeps in tests for pagination and inputs. In Shamir recovery tests that's 20s per mnemonic, which taken together is A Lot(tm)

i left the shamir tests out of this PR in order not to conflict with #358 

@jpochyla plz review, i am VERY unsure about the `feedback_signal` class

### CODE REVIEW

**Comment by jpochyla** [comment]
Please rebase on top of master.

**Comment by matejcik** [comment]
The approach turns out to be flawed.

It will work fine if the only feature required is "wait until core loop picks up the DebugLinkDecision value" -- this helps with entering mnemonic words, so it might be worth it anyway.

In other cases, however, it won't help. Namely, the debuglink decision returns *after* swipe command is consumed but *before* it is processed. Similar thing happens with the "reset word" screen. A different approach is required and I'm not sure what.
Pagination can be solved with explicit callback from the `Paginate` class. Note that in many tests we don't actually need to scroll through pages; the whole dialog accepts `confirm` signals. In these cases perhaps input flows are not necessary anymore.
However, it seems that the reset-word-index screen exhibits the same problem. It's unclear what to do there. It would be difficult to hook into the same mechanism.

**Comment by matejcik** [comment]
reimplemented the thing with channels.

i also included a fix for #373, and shortened the Shamir tests

**Comment by jpochyla** [comment]
❤️

File: core/src/trezor/loop.py [hunk 1]
```diff
@@ -343,3 +296,87 @@ def __iter__(self) -> Task:  # type: ignore
             # close() or throw(), kill the children tasks and re-raise
             self.exit()
             raise
+
+
+class chan:
+    """
```
Comment (tsusanka) [inline core/src/trezor/loop.py]: It has docs :heart:

**Pull request merged**